website/static/js/logFeed.js

```diff
@@ -1,238 +1,94 @@
-            nodePath: item.node.path,
-            user: item.user,
-            registrationCancelled: item.node.is_registration && item.node.registered_from_id == null
-        });
-    });
-    return mappedLogs;
-};
-
-////////////////
-// Public API //
-////////////////
+            function _processResults (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                    if(addToExistingList){
+                        self.activityLogs().push(log);
+                    }
+                });
+                if(!addToExistingList){
+                    self.activityLogs(result.data);  // Set activity log data
+                }
+                self.showMoreActivityLogs(result.links.next); // Set view for show more button
+            }
 
-var defaults = {
-    /** Selector for the progress bar. */
-    progBar: '#logProgressBar'
-};
+            self.logRequestPending = true;
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            promise.then(_processResults);
+            promise.then(function(){
+                self.logRequestPending = false;
+                return promise;
+            });
+        };
+
+        self.getCurrentLogs = function _getCurrentLogs (node){
+            if(!self.logRequestPending) {
+                if (!node.is_retracted) {
+                    var urlPrefix = node.is_registration ? 'registrations' : 'nodes';
+                    var query = node.link ? { 'page[size]': 6, 'embed': ['original_node', 'user', 'linked_node', 'template_node'], 'view_only': node.link} : { 'page[size]': 6, 'embed': ['original_node', 'user', 'linked_node', 'template_node']};
+                    var url = $osf.apiV2Url(urlPrefix + '/' + node.id + '/logs/', { query: query});
+                    var promise = self.getLogs(url);
+                    return promise;
+                }
+            }
+        };
 
+        self.getCurrentLogs(self.node);
+    },
 
-var initViewModel = function(self, logs, url){
-    self.logs = createLogs(logs);
-    self.viewModel = new LogsViewModel(self.logs, url);
-    if(url) {
-        self.viewModel.fetchResults();
+    view : function (ctrl) {
+        return m('.db-activity-list.m-t-md', [
+            ctrl.activityLogs() ? ctrl.activityLogs().map(function(item){
+                if (ctrl.node.anonymous) { item.anonymous = true; }
+                var image = m('i.fa.fa-desktop');
+                if (!item.anonymous && item.embeds.user && item.embeds.user.data) {
+                    image = m('img', { src : item.embeds.user.data.links.profile_image});
+                } else if (!item.anonymous && item.embeds.user && item.embeds.user.errors[0].meta){
+                    image = m('img', { src : item.embeds.user.errors[0].meta.profile_image});
+                }
+                return m('.db-activity-item', [
```

Comment:
The sheer number of nested ternaries/conditionals in this would usually be a red flag, but given the complexity of what this is doing, it may be acceptable. I'll leave the final word on that up to @sloria